src/storage/garbage_collector.cpp

```diff
@@ -78,7 +79,10 @@ uint32_t GarbageCollector::ProcessUnlinkQueue() {
   while (!txns_to_unlink_.empty()) {
     txn = txns_to_unlink_.front();
     txns_to_unlink_.pop_front();
-    if (txn->undo_buffer_.Empty()) {
+
+    if (txn->IsReadOnly()) {
+      // TODO(Tianyu): Deal with this edge case
```

Comment:
TODO in this PR?

Reply:
This is stale as I fixed it using the deferred event framework.